nvme-print-json: Fix linux kernel check patch errors

ikegami-t opened the pull request:
Note: The long line string and strncpy() warnings are not fixed.

ikegami-t commented:
Depended on the libnvme PR changes https://github.com/linux-nvme/libnvme/pull/828.

igaw commented on nvme-print-json.c:
```diff
@@ -69,15 +69,15 @@ static void obj_add_uint_0x(struct json_object *o, const char *k, __u32 v)
 	char str[STR_LEN];
 
-	sprintf(str, "0x%02x", v);
+	sprintf(str, "%#0*x", v ? 4 : 2, v);
```
This looks a bit wierd. The idea of this function is to print always two digits with the prefix `0x`. 

```c
        v = 0;
	printf("%#0*x\n", v ? 4 : 2, v);
	v = 456;
	printf("%#0*x\n", v ? 4 : 2, v);
```

gives

```
00
0x1c8
```

ikegami-t replied:
In the past a related change done by the commit 4c7b19a also since `0x` included as field length as below.
```c
	v = 1;
	printf("%#0*x\n", v ? 4 : 2, v);
```
gives
```
0x1
```
but expected
```
0x01
```

igaw replied:
after playing around with `#` I think we should just stick with prefixing the string ourself. Too much troubles for something so trivial and I really would like to avoid to have these `v ? 4 : 2` around.

ikegami-t commented:
Rebased the PR with latest master and fixed the issue mentioned by the comment https://github.com/linux-nvme/nvme-cli/pull/2321#discussion_r1589071418.

igaw commented:
Thanks!
